azkaban-common/src/main/java/azkaban/event/EventHandler.java

```diff
@@ -16,28 +16,33 @@
 
 package azkaban.event;
 
-import java.util.ArrayList;
 import java.util.HashSet;
 
-public class EventHandler {
-  private HashSet<EventListener> listeners = new HashSet<EventListener>();
+/**
+ * Due to History legacy, this interface is being used by many other modules, and we can not
+ * entirely refactor this interface.
+ * Having default interface methods helps implementation classes easily write necessary logics.
+ *
+ * TODO: Refactor return type in getListeners method.
+ */
+public interface EventHandler {
 
-  public EventHandler() {
-  }
+  HashSet<EventListener> getListeners();
```

Comment:
The common convention is to have `Interface I = new Implementation();` so probably `Set<EventListener>` instead of `HashSet` ?